pkg/typesystem/error.go

```diff
@@ -9,66 +8,100 @@ import (
 )
 
 var (
-	ErrModelNotFound          = errors.New("authorization model not found")
-	ErrDuplicateTypes         = errors.New("an authorization model cannot contain duplicate types")
-	ErrInvalidSchemaVersion   = errors.New("invalid schema version")
-	ErrInvalidModel           = errors.New("invalid authorization model encountered")
-	ErrRelationUndefined      = errors.New("undefined relation")
-	ErrObjectTypeUndefined    = errors.New("undefined object type")
-	ErrInvalidUsersetRewrite  = errors.New("invalid userset rewrite definition")
-	ErrReservedKeywords       = errors.New("self and this are reserved keywords")
-	ErrCycle                  = errors.New("an authorization model cannot contain a cycle")
-	ErrNoEntrypoints          = errors.New("no entrypoints defined")
-	ErrNoEntryPointsLoop      = errors.New("potential loop")
+	// ErrModelNotFound is returned when an authorization model is not found.
+	ErrModelNotFound = errors.New("authorization model not found")
+
+	// ErrDuplicateTypes is returned when an authorization model contains duplicate types.
+	ErrDuplicateTypes = errors.New("an authorization model cannot contain duplicate types")
+
+	// ErrInvalidSchemaVersion is returned for an invalid schema version in the authorization model.
+	ErrInvalidSchemaVersion = errors.New("invalid schema version")
+
+	// ErrInvalidModel is returned when encountering an invalid authorization model.
+	ErrInvalidModel = errors.New("invalid authorization model encountered")
+
+	// ErrRelationUndefined is returned when encountering an undefined relation in the authorization model.
+	ErrRelationUndefined = errors.New("undefined relation")
+
+	// ErrObjectTypeUndefined is returned when encountering an undefined object type in the authorization model.
+	ErrObjectTypeUndefined = errors.New("undefined object type")
+
+	// ErrInvalidUsersetRewrite is returned for an invalid userset rewrite definition.
+	ErrInvalidUsersetRewrite = errors.New("invalid userset rewrite definition")
+
+	// ErrReservedKeywords is returned when using reserved keywords "self" and "this".
+	ErrReservedKeywords = errors.New("self and this are reserved keywords")
+
+	// ErrCycle is returned when an authorization model contains a cycle.
```

Comment:
```suggestion
	// ErrCycle is returned when an authorization model contains a cycle because an objectType and relation in a model defines a rewrite rule that is self-referencing through computed relationships
```

Example 

<img width="934" alt="image" src="https://github.com/openfga/openfga/assets/5374887/31313999-d426-42b3-945e-a1c054272044">

Reply:
Much appreciated @miparnisari 🙇🏻 the updated explanations with the graphs are extremely valuable in all your comments. Helps me understand a lot better the domain. 

Everything was accounted for within [`4446f6a` (#1264)](https://github.com/openfga/openfga/pull/1264/commits/4446f6a1c9da5d7d08b421d7477182cb817c9610).

Reply:
@sergiught it actually made me think that `ErrCycle` and `ErrNoEntryPointsLoop` represent the same error 😆 i don't know why we have two 🤔